### ISSUE TITLE
System.InvalidCastException when deserializing a Dictionary<string, bool>

### ISSUE THREAD

**Issue by Bond-009** [paiden/Nett#34]
```
Unhandled Exception: System.InvalidCastException: Unable to cast object of type 'System.Boolean' to type 'System.String'.
   at Nett.TomlConverterBase`2.Convert(ITomlRoot root, Object o, Type targetType) in C:\dev\nett\Source\Nett\TomlConverterBase.cs:line 23
   at Nett.TomlObject.<>c__DisplayClass18_0.<CreateArrayType>b__0(Object o) in C:\dev\nett\Source\Nett\TomlObject.cs:line 93
   at Nett.LinqExtensions.EnumerableExtensions.<Select>d__1`1.MoveNext() in C:\dev\nett\Source\Nett\Extensions\EnumerableExtensions.cs:line 31
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Nett.TomlObject.CreateArrayType(ITomlRoot root, IEnumerable e) in C:\dev\nett\Source\Nett\TomlObject.cs:line 94
   at Nett.TomlTable.From[T](TomlSettings settings, T obj) in C:\dev\nett\Source\Nett\TomlTable.RootTable.cs:line 27
```

**Comment by paiden** [comment]
Could you provide a small repro example?

**Comment by Bond-009** [comment]
code:
```cs
using System;
using System.Collections.Generic;
using Nett;

namespace Nett_crash
{
    class Program
    {
        static void Main(string[] args)
        {
            Toml.WriteString(new Dictionary<string, bool>(){
                {"Hello World!", true}
            });
        }
    }
}
```

Exception:
```
Unhandled Exception: System.InvalidCastException: Unable to cast object of type 'System.Boolean' to type 'System.String'.
   at Nett.TomlConverterBase`2.Convert(ITomlRoot root, Object o, Type targetType) in C:\dev\nett\Source\Nett\TomlConverterBase.cs:line 23
   at Nett.TomlObject.<>c__DisplayClass18_0.<CreateArrayType>b__0(Object o) in C:\dev\nett\Source\Nett\TomlObject.cs:line 93
   at Nett.LinqExtensions.EnumerableExtensions.<Select>d__1`1.MoveNext() in C:\dev\nett\Source\Nett\Extensions\EnumerableExtensions.cs:line 31
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Nett.TomlObject.CreateArrayType(ITomlRoot root, IEnumerable e) in C:\dev\nett\Source\Nett\TomlObject.cs:line 94
   at Nett.TomlTable.From[T](TomlSettings settings, T obj) in C:\dev\nett\Source\Nett\TomlTable.RootTable.cs:line 27
   at Nett_crash.Program.Main(String[] args) in .../Nett_crash/Program.cs:line 11
```

**Comment by paiden** [comment]
Writing a dictionary directly is not supported at the moment. This method can only write TomlTabe and custom CLR objects currently. 

I will add a default converter for this case in the future.

**Comment by paiden** [comment]
Fixed in v0.8.0
